src/django_upgrade/fixers/test_http_headers.py

```diff
@@ -0,0 +1,189 @@
+"""
+Transforms HTTP headers from WSGI kwarg format to new 'headers' dictionary, for
+test Client and RequestFactory:
+https://docs.djangoproject.com/en/4.2/releases/4.2/#tests
+"""
+from __future__ import annotations
+
+import ast
+from functools import partial
+from typing import Iterable
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from django_upgrade.ast import ast_start_offset
+from django_upgrade.compat import str_removeprefix
+from django_upgrade.data import Fixer
+from django_upgrade.data import State
+from django_upgrade.data import TokenFunc
+from django_upgrade.fixers.assert_form_error import looks_like_client_call
+from django_upgrade.tokens import find
+from django_upgrade.tokens import NAME
+from django_upgrade.tokens import OP
+from django_upgrade.tokens import parse_call_args
+
+fixer = Fixer(
+    __name__,
+    min_version=(4, 2),
+)
+
+HEADERS_KWARG = "headers"
+HTTP_PREFIX = "HTTP_"
+
+
+@fixer.register(ast.Call)
+def visit_Call(
+    state: State,
+    node: ast.Call,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        isinstance(node.func, ast.Name)
+        and node.func.id in ("Client", "RequestFactory")
+        and node.func.id in state.from_imports["django.test"]
+    ) or (looks_like_client_call(node, "client") and node.args):
```

Comment:
filter for keyword argument here, to avoid queueing a token rewriting functions for every client call, on every run